utils.go

```diff
@@ -42,7 +42,7 @@ func isTag(event, ref string) bool {
 
 // helper function to write a netrc file.
 func writeNetrc(machine, login, password string) error {
-	if machine == "" {
+	if machine == "" || (login == "" && password == "") {
```

Comment:
Do we have a case where only login or password is provided and the other value ist still empty?

Otherwise we should use:

```suggestion
	if machine == "" || login == "" || password == "" {
```

Reply:
What if `login` or `password` is a token while other is empty?

Reply:
Okay, we can leave it as is for now.